test/controllers/AddClosureMetadataControllerSpec.scala

```diff
@@ -180,3 +185,17 @@ class AddClosureMetadataControllerSpec extends FrontEndTestHelper {
             Values("mock code1", List()), Values("mock code2", List())))))
   }
 }
+
+case class MockAsyncCacheApi()(implicit val ec: ExecutionContext) extends AsyncCacheApi {
```

Comment:
I'm not sure if this will be useful later on so feel free to ignore me but you don't need a full case class here, you can use Mockito to pass in a mocked `AsyncCacheApi` so the controller constructor becomes
```scala
    new AddClosureMetadataController(
     ...
      customMetadataService,
      mock[AsyncCacheApi]
    )
```
If you do need to mock the responses from calls to the cache API then that can be done.

Reply:
Great! Thanks.